hpx/parallel/datapar/zip_iterator.hpp

```diff
@@ -0,0 +1,194 @@
+//  Copyright (c) 2016 Hartmut Kaiser
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(HPX_PARALLEL_DATAPAR_ZIP_ITERATOR_SEP_30_2016_1251PM)
+#define HPX_PARALLEL_DATAPAR_ZIP_ITERATOR_SEP_30_2016_1251PM
+
+#include <hpx/config.hpp>
+
+#if defined(HPX_HAVE_DATAPAR)
+
+#include <hpx/util/detail/pack.hpp>
+#include <hpx/util/zip_iterator.hpp>
+#include <hpx/traits/is_iterator.hpp>
+
+#include <hpx/parallel/datapar/iterator_helpers.hpp>
+#include <hpx/parallel/traits/vector_pack_alignment_size.hpp>
+#include <hpx/parallel/traits/vector_pack_load_store.hpp>
+#include <hpx/parallel/traits/vector_pack_type.hpp>
+
+#include <algorithm>
+#include <cstddef>
+#include <iterator>
+#include <memory>
+#include <numeric>
+#include <type_traits>
+
+///////////////////////////////////////////////////////////////////////////////
+namespace hpx { namespace parallel { namespace util { namespace detail
+{
+    ///////////////////////////////////////////////////////////////////////////
+    template <typename ... Iter>
+    struct is_data_aligned_impl<hpx::util::zip_iterator<Iter...> >
+    {
+        template <std::size_t ... Is>
+        static HPX_FORCEINLINE bool
+        call(hpx::util::zip_iterator<Iter...> const& it,
+            hpx::util::detail::pack_c<std::size_t, Is...>)
+        {
+            auto const& t = it.get_iterator_tuple();
+            int const sequencer[] = {
+                0, (is_data_aligned(hpx::util::get<Is>(t)), 0)...
```

Comment:
Yah, we could use `std::any_of()` over a sequence of `bool`s. That might even be (slightly) faster.